boxsdk/session/box_session.py

```diff
@@ -376,6 +378,8 @@ def _prepare_and_send_request(
                 break
 
             attempt_number += 1
+            network_response.log(can_safely_log_content=True)
```

Comment:
Looks like this partially addresses issue #190 .

As another idea, what if we put this into the constructor for `DefaultNetworkResponse`.

```python
def __init__(self, request_response, access_token_used):
    ....
    if not self.ok:
        self.log(can_safely_log_content=True)
```

(or we could put equivalent code into `DefaultNetwork.request()`)

The idea being that, if the request fails, then in theory the content is never going to be streamed (we could even enforce that if we wanted).  And so in those cases, it is safe to log immediately, without waiting to see how the caller is going to access the response data.

In theory, this would completely address #190.